src/test/java/teammates/test/cases/browsertests/InstructorFeedbackEditPageUiTest.java

```diff
@@ -300,6 +303,63 @@ private void testAddQuestionAction() throws Exception {
         feedbackEditPage.verifyHtmlMainContent("/instructorFeedbackQuestionAddSuccess.html");
     }
 
+    private void testAddTemplateQuestionLink() throws Exception {
+
+        ______TS("add template question link");
+
+        feedbackEditPage.clickNewQuestionButton();
+        feedbackEditPage.clickTemplateModalButton();
+
+        assertFalse("Unable to submit when there is no question selected",
+                feedbackEditPage.isAddTemplateQuestionButtonEnabled());
+    }
+
+    private void testAddTemplateQuestionAction() throws Exception {
+
+        ______TS("add template question action success");
+
+        // check if each question panel is expandable
+        feedbackEditPage.clickTemplateQuestionPanel(1);
+        feedbackEditPage.clickTemplateQuestionPanel(2);
+        feedbackEditPage.clickTemplateQuestionPanel(3);
+        feedbackEditPage.clickTemplateQuestionPanel(4);
+        feedbackEditPage.clickTemplateQuestionPanel(5);
+        feedbackEditPage.verifyHtmlPart(By.id("addTemplateQuestionModal"),
+                "/instructorFeedbackTemplateQuestionExpandedModal.html");
+
+        feedbackEditPage.clickTemplateQuestionModalCheckBox(1);
+
+        assertTrue("Can click after selecting a question",
+                feedbackEditPage.isAddTemplateQuestionButtonEnabled());
+        feedbackEditPage.clickAddTemplateQuestionButton();
+
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_QUESTION_ADDED);
+        assertNotNull(getFeedbackQuestionWithRetry(courseId, feedbackSessionName, 2));
+
+        feedbackEditPage.verifyHtmlMainContent("/instructorFeedbackTemplateQuestionAddSuccess.html");
+
+        ______TS("add multiple template questions success");
+
+        feedbackEditPage.clickNewQuestionButton();
+        feedbackEditPage.clickTemplateModalButton();
+
+        feedbackEditPage.clickTemplateQuestionModalCheckBox(2);
+        feedbackEditPage.clickTemplateQuestionModalCheckBox(3);
+        feedbackEditPage.clickAddTemplateQuestionButton();
+
+        feedbackEditPage.waitForTextsForAllStatusMessagesToUserEquals(
+                Const.StatusMessages.FEEDBACK_QUESTION_ADDED_MULTIPLE);
+        assertNotNull(getFeedbackQuestionWithRetry(courseId, feedbackSessionName, 4));
+
+        // revert back to state expected by tests after this by deleting newly added template questions
+        String questionId = getFeedbackQuestionWithRetry(courseId, feedbackSessionName, 4).getId();
+        BackDoor.deleteFeedbackQuestion(questionId);
+        questionId = getFeedbackQuestionWithRetry(courseId, feedbackSessionName, 3).getId();
+        BackDoor.deleteFeedbackQuestion(questionId);
+        questionId = getFeedbackQuestionWithRetry(courseId, feedbackSessionName, 2).getId();
+        BackDoor.deleteFeedbackQuestion(questionId);
```

Comment:
Is it possible to reorder the tests so that we dont have to do such things as they take up time and resource?

Reply:
As I am trying to test adding multiple template questions so total 3 questions will be added to the session to ensure that it works. I will need to delete them else the `verifyHtmlContent` of other files will increase and that will require time as well. 

Alternatively, I can call `deleteQuestionLink` and `waitForConfirmationModalAndClickOk` to remove the 3 questions but I think that will take longer time right?

The only way I can think of is to move the `testAddTemplateQuestionLink` and `testAddTemplateQuestionAction` functions to the last functions under `testGeneralQuestionOperations`. Hence, I do not need to delete the questions as the session will be deleted in `testDeleteSessionAction` thereafter.

Reply:
> The only way I can think of is to move the testAddTemplateQuestionLink and testAddTemplateQuestionAction functions to the last functions under testGeneralQuestionOperations. Hence, I do not need to delete the questions as the session will be deleted in testDeleteSessionAction thereafter.

This sounds good cause then we dont access the datastore or waste time in manually deleting